demo/config/targets/TEMPLATED/lib/templated_interface.rb

```diff
@@ -28,7 +28,11 @@ def connect
         @sleeper = Sleeper.new
         @polling_thread = Thread.current
         begin
-          while connected?
+          # Wait for the connection to actually occur
+          while !connected?
+            @sleeper.sleep(1)
```

Comment:
We have to wait for the connection to occur now instead of having it be immediately connected.